build.gradle

```diff
@@ -14,14 +14,27 @@ plugins {
 	id 'org.sonarqube' version '2.8'
 	id 'checkstyle'
 	id 'org.ajoberstar.grgit' version '4.0.1'
+	id 'com.jfrog.artifactory' version '4.14.1' apply false
 	id 'io.spring.dependency-management' version '1.0.9.RELEASE'
 }
 
 description = 'Spring Integration Kafka Support'
 
+apply plugin: 'java-library'
 apply plugin: 'kotlin'
 apply plugin: 'kotlin-spring'
 
+java {
+	withJavadocJar()
+	withSourcesJar()
+	registerFeature('optional') {
```

Comment:
Looks like we don't have any `optional` and `provided`, so, probably we can live without this features.